src/liblzma/check/crc64_fast.c

```diff
@@ -438,35 +438,47 @@ is_clmul_supported(void)
 }
 
 
-#ifdef HAVE_FUNC_ATTRIBUTE_CONSTRUCTOR
-#	define CRC64_FUNC_INIT
-#	define CRC64_SET_FUNC_ATTR __attribute__((__constructor__))
-#else
-#	define CRC64_FUNC_INIT = &crc64_dispatch
-#	define CRC64_SET_FUNC_ATTR
-static uint64_t crc64_dispatch(const uint8_t *buf, size_t size, uint64_t crc);
-#endif
+#ifdef HAVE_FUNC_ATTRIBUTE_IFUNC
+#	ifdef __cplusplus
+// The extern variable "C" is required to work with C++'s function name 
+// mangling in C++. If this is not included, the "crc64_resolver" will be
+// mangled and won't compile.
+extern "C" {
+#	endif
 
+// The resolver must accept no arguments (void), and must specify the return and
+// parameter types in order to avoid warnings. The resolved function return type
+// appears where a regular function's return type would be, and the resolved
+// functions parameters appear after the resolver's arguments (void).
+static uint64_t
+(*crc64_resolver(void)) (const uint8_t*, size_t, uint64_t)
+{
+	return is_clmul_supported() ? crc64_clmul : crc64_generic;
+}
+
+#	ifdef __cplusplus
+}
+#	endif
 
-// Pointer to the the selected CRC64 method.
-static uint64_t (*crc64_func)(const uint8_t *buf, size_t size, uint64_t crc)
-		CRC64_FUNC_INIT;
+// The resolved function can be called as crc64_func().
+static uint64_t crc64_func(const uint8_t* buf, size_t size, uint64_t crc)
```

Comment:
To keep the style consistent, please change the pointer declarations to be type *var_name instead of type* var_name